h2/utilities.py

```diff
@@ -298,22 +324,94 @@ def _check_host_authority_header(headers, hdr_validation_flags):
 
         yield header
 
-    # If we have not-None values for these variables, then we know we saw
-    # the corresponding header.
-    authority_present = (authority_header_val is not None)
-    host_present = (host_header_val is not None)
+    _validate_host_authority_header(
+        authority_header_val, host_header_val, hdr_validation_flags)
 
-    # It is an error for a request header block to contain neither
-    # an :authority header nor a Host header.
-    if not authority_present and not host_present:
-        raise ProtocolError(
-            "Did not receive an :authority or Host header."
-        )
 
-    # If we receive both headers, they should definitely match.
-    if authority_present and host_present:
-        if authority_header_val != host_header_val:
-            raise ProtocolError(
-                "Received mismatched :authority and Host headers: %r / %r" %
-                (authority_header_val, host_header_val)
-            )
+def _secure_headers(headers, hdr_validation_flags):
```

Comment:
I moved this down just so all the sent-headers pipeline functions are nicely grouped together.